docs/find-duplicate-skus/script.liquid

```diff
@@ -0,0 +1,103 @@
+{% assign mappings = hash %}
+{% assign dups = hash %}
+{% assign cursor = nil %}
+
+{% for i in (0..500) %}
+  {% assign after = i | times: 250 %}
+  {% capture query %}
+    query {
+      productVariants(
+        first: 250
+        after: {{ cursor | json }}
+      ) {
+        pageInfo {
+          hasNextPage
+        }
+        edges {
+          node {
+            id
+            sku
+            inventoryItem {
+              tracked
+            }
+          }
+          cursor
+        }
+      }
+    }
+  {% endcapture %}
+
+  {% assign result = query | shopify %}
+
+  {% if event.preview %}
+    {% capture result_json %}
+      {
+        "data": {
+          "productVariants": {
+            "edges": [
+              {
+                "node": {
+                  "id": "gid://shopify/ProductVariant/1234567890",
+                  "sku": "sku123",
+                  "inventoryItem": {
+                    "tracked": "true"
+                  }
+                }
+              },
+              {
+                "node": {
+                  "id": "gid://shopify/ProductVariant/123456789",
+                  "sku": "sku123",
+                  "inventoryItem": {
+                    "tracked": "true"
+                  }
+                }
+              }
+            ]
+          }
+        }
+      }
+    {% endcapture %}
+
+    {% assign result = result_json | parse_json %}
+  {% endif %}
+  
+  {% assign cursor = result.data.productVariants.edges.last.cursor %}
+  {% assign productVariants = result.data.productVariants.edges | map: "node" %}
+
+  {% for pv in productVariants %}
+    {% if pv.sku == null or pv.sku == empty %}
+      {% continue %}
+    {% endif %}
+
+    {% if options.exclude_inventory_not_tracked__boolean == true %}
+      {% if pv.inventoryItem.tracked == false %}
+        {% continue %}
+      {% endif %}
+    {% endif %}
+
+    {% assign existingPv = mappings[pv.sku] %}
+
+    {% if existingPv %}
+      {% if existingPv != pv.id %}
+        {% assign dups[pv.sku] = true %}
+      {% endif %}
+    {% else %}
+      {% assign mappings[pv.sku] = pv.id %}
+    {% endif %}
+  {% endfor %}
+
+  {% unless result.data.productVariants.pageInfo.hasNextPage %}
+    {% break %}
+  {% endunless %}
+{% endfor %}
+
+{% assign skus = productVariants | map: "sku" %}
+{% assign uniqskus = skus | uniq %}
+
+{% if skus.size == uniqskus.size %}
+  {% comment %}No exact duplicate SKUS {% endcomment %}
+  {% break %}
```

Comment:
I ... am super super surprised that this works, haha. I had no idea `{% break %}` could be used this way. _Super_ useful - thank you! :D

Reply:
For my own edification and others viewing this PR, what exactly is the purpose of this break, since there is no for loop to break out of in the code at this juncture?

Reply:
It appears that `{% break %}` (and `{% continue %}`, actually) are considered interrupts _anywhere_ they occur - not just in forloops. Using it here is like returning early.

Fascinating. :D

Related reading:
* a break is an interrupt: https://github.com/Shopify/liquid/blob/eab13a07d9861a38d993d2749ae25f06ff76426b/lib/liquid/tags/break.rb
* recording interrupts: https://github.com/Shopify/liquid/blob/eab13a07d9861a38d993d2749ae25f06ff76426b/lib/liquid/context.rb#L76-L84
* reacting to an interrupt: https://github.com/Shopify/liquid/blob/eab13a07d9861a38d993d2749ae25f06ff76426b/lib/liquid/block_body.rb#L214-L217